tests/deployment_tests/gitlab.js

```diff
@@ -0,0 +1,201 @@
+  } catch (error) {
+    // this error will happen when the https call to gitlab fails
+    throw new Error('Error in listing users');
+  }
+  return usernames;
+}
+
+class User {
+  constructor(username, password) {
+    /* Default password if set to 12345678 */
+    this.username = username;
+    this.password = password || 12345678;
+  }
+
+  async addUser() {
+    const emailId = `${this.username}@autolab.com`;
+    let body = {};
+    try {
+      const token = await getToken();
+      body = await request.post(`https://${gitlabHostname}/api/v4/users?private_token=${token}`, {
+        json: {
+          name: this.username,
+          username: this.username,
+          password: this.password,
+          email: emailId,
+          skip_confirmation: true,
+        },
+      });
+    } catch (error) {
+      body = error.error;
+      if (error.message !== '409 - {"message":"Email has already been taken"}') {
+        throw new Error(`Error in adding user ${this.username}`);
+      }
+    }
+    return body;
+  }
+
+  async deleteUser() {
+    let body;
+    try {
+      const token = await getToken();
+      const userId = await getId(this.username);
+      body = await request.delete(`https://${gitlabHostname}/api/v4/users/${userId}?private_token=${token}`, { json: { id: userId, hard_delete: true } });
+    } catch (error) {
+      body = error.error;
+      if (error.message !== '404 - {"error":"404 Not Found"}') {
+        throw new Error(`Error in deleting user ${this.username}`);
+      }
+    }
+    return body;
+  }
+}
+
+class Project {
+  constructor(username, password, projectName) {
+    this.username = username;
+    this.password = password || 12345678;
+    this.projectName = projectName;
+  }
+
+  async createProject() {
+    let body;
+    try {
+      const token = await getToken(this.username, this.password);
+      body = await request.post(`https://${gitlabHostname}/api/v4/projects?private_token=${token}`, { json: { name: this.projectName } });
+    } catch (error) {
+      body = error.error;
+      if ((!(error.message).includes('400 - {"message":{"route.path":["has already been taken"]'))) {
+        throw new Error(`Error in creating project ${this.projectName} for user ${this.username}`);
+      }
+    }
+    return body;
+  }
+
+  async commit(commitFilesPath) {
+    const projectPath = `https://${this.username}:${this.password}@${gitlabHostname}/${this.username}/${this.projectName}.git`;
+    let commitDetails;
+    try {
+      await simpleGit.cwd(commitFilesPath);
+      await simpleGit.init();
+      await simpleGit.add(`${commitFilesPath}/*`);
+      commitDetails = await simpleGit.commit('first commit!');
+      await simpleGit.addRemote('origin', projectPath);
+      await simpleGit.push('origin', 'master');
+    } catch (error) {
+      throw new Error('Commit failed');
+    }
+    return commitDetails;
+  }
+}
+
+
+/*
+The options allowed are:
```

Comment:
The deployment tests are to be automated. User interaction is a hindrance there. Please remove the interaction part completely.